packages/opencensus-exporter-zpages/src/zpages-frontend/page-handlers/tracez.page-handler.ts

```diff
@@ -0,0 +1,122 @@
+
+const ejs = require('ejs');
+const timeunit = require('time-unit');
+
+export type TraceConfigzParams = {
+  tracename: string;
+};
+
+export class TracezPageHandler {
+  constructor() {}
+
+  /**
+   * Generate Zpages Tracez HTML Page
+   * @param params values to put in HTML template
+   * @returns Output HTML
+   */
+  emitHtml(params?: TraceConfigzParams): string {
+    const tracezFile =
+        ejs.fileLoader(__dirname + '/../templates/tracez.ejs', 'utf8');
+    const summaryFile =
+        ejs.fileLoader(__dirname + '/../templates/summary.ejs', 'utf8');
+    const spanCellFile =
+        ejs.fileLoader(__dirname + '/../templates/span-cell.ejs', 'utf8');
+    const spansFile =
+        ejs.fileLoader(__dirname + '/../templates/spans.ejs', 'utf8');
+    /** ejs render options */
+    const options = {delimiter: '?'};
+    /** latency array */
+    const latencyBucketBoundaries = LatencyBucketBoundaries.values;
+    /** zpages exporter singleton instance */
+    const zpages = tracing.exporter as ZpagesExporter;
+    /** trace list */
+    const traces = zpages.getAllTraces();
+    /** latency names list */
+    const latencyBucketNames = [];
+    /** rootSpan lines */
+    let spansCells = '';
+    /** span list */
+    let spans = '';
+    /** control the striped table lines */
+    let stripedCell = true;
+
+    // build a string list of latencyBucketBoundaries
+    for (const latency of latencyBucketBoundaries) {
+      latencyBucketNames.push(`[${latency}]`);
+    }
+
+    // build the table lines
+    for (const spanName of Object.keys(traces).sort()) {
+      const spans = traces[spanName];
+      const spanCell = {
+        spanCellClass: stripedCell ? 'striped' : '',
+        name: spanName,
+        RUNNING: 0,
+        ZERO_MICROSx10: 0,
+        MICROSx10_MICROSx100: 0,
+        MICROSx100_MILLIx1: 0,
+        MILLIx1_MILLIx10: 0,
+        MILLIx10_MILLIx100: 0,
+        MILLIx100_SECONDx1: 0,
+        SECONDx1_SECONDx10: 0,
+        SECONDx10_SECONDx100: 0,
+        SECONDx100_MAX: 0,
+        errors: 0
+      };
+
+      // build span list
+      for (const span of spans) {
+        if (span.ended) {
+          const durationNs = timeunit.milliseconds.toNanos(span.duration);
+          for (const latency of latencyBucketBoundaries) {
+            if (latency.belongs(durationNs)) {
+              spanCell[latency.getName()] += 1;
+              break;
+            }
+          }
+        } else if (span.started) {
+          spanCell['RUNNING'] += 1;
+        }
+      }
+
+      spansCells += ejs.render(spanCellFile, spanCell, options);
+      stripedCell = !stripedCell;
+    }
+
+    /** HTML table summary */
+    const summary = ejs.render(summaryFile, {latencyBucketNames}, options);
+    if (params.tracename) {
+      spans = ejs.render(
+          spansFile, {
+            name: params.tracename,
+            traces: zpages.getTracesByName(params.tracename)
```

Comment:
In my opinion `getTracesByName` should not be on the `ZpagesExporter` interface, because it's not a feature of Zpages. It's possible to get these traces by sharing the value of `traces` that is a private member in `ZpagesExporter` (by passing it to the constructor of `TracezPageHandler`).

I understand that `getTracesByName` are also used in tests but arguably that is not really testing the exporter's behavior. The behavior that should be tested is that the the server, when queried, returns information about current and recent spans. Ideally, the server could have "JSON" format endpoints so that we don't have to re-parse the HTML in tests to check this behavior.

This can be a `TODO` -- but I think we should remove this from the public API.

Reply:
Ok, I understood. I removed the ```getTracesByName``` and I'm adding a TODO to create the JSON endpoint.